### PR TITLE
Add typehints and correcting stylistic errors

### PR DESCRIPTION
**Pull request by ostrokach**
Goes together with: pyta-uoft/website#67

### CODE REVIEW

File: examples/assigning_to_self_example.py [hunk 1]
```diff
@@ -1,10 +1,10 @@
 class Assigning():
```
Comment (david-yz-liu) [inline examples/assigning_to_self_example.py]: @ostrokach One other stylistic matter - please omit the `()` for class definitions.

File: examples/global_variables_example.py [hunk 2]
```diff
@@ -1,37 +1,32 @@
-def function():
-    """ A test for the global variables checker
-    @rtype: None
-    """
+def function() -> None:
+    """ A test for the global variables checker."""
```
Comment (david-yz-liu) [inline examples/global_variables_example.py]: Please remove the space between the open `"""` and the text.

File: examples/pylint/C0121_singleton_comparison.py [hunk 3]
```diff
@@ -1,4 +1,6 @@
-def square(number):
+from typing import Optional
+
+def square(number: float) -> Optional[float]:
```
Comment (david-yz-liu) [inline examples/pylint/C0121_singleton_comparison.py]: Shouldn't the parameter's type also be `Optional[float]`?

File: examples/pylint/C0321_multiple_statements.py [hunk 4]
```diff
@@ -1,8 +1,3 @@
-def pos(temp):
-    """Return 'positive' if <temp> is greater than 0. Else return 'negative'.
-    @type temp: int
-    @rtype: str
-    """
+def pos(temp: int) -> str:
     if temp > 0: return 'positive'
-    else:
-        return 'negative'
+    return 'negative'
```
Comment (david-yz-liu) [inline examples/pylint/C0321_multiple_statements.py]: Just a reminder to revert this change.

File: examples/pylint/R0915_too_many_statements.py [hunk 5]
```diff
@@ -1,9 +1,11 @@
-def statement(arg):
-    """Dummy function to demonstrate an example within too_many_statements()"""
+from typing import Any
+
+def statement(arg: Any) -> None:
+    """Dummy function to demonstrate an example within too_many_statements()."""
```
Comment (david-yz-liu) [inline examples/pylint/R0915_too_many_statements.py]: Please remove the `()`.

File: examples/pylint/W1303_missing_format_argument_key.py [hunk 6]
```diff
@@ -1,2 +1,3 @@
 # Error on the following line: missing format argument for 'age'
-s = '{last_name}, {fist_name} - {age}'.format(last_name='bond', first_name='james')
+s = '{last_name}, {fist_name} - {age}'.format(
+    last_name='bond', first_name='james')
```
Comment (david-yz-liu) [inline examples/pylint/W1303_missing_format_argument_key.py]: Is this one really the correct indentation? I'm a little skeptical.

File: examples/pylint/C0321_multiple_statements.py (hunk 4 above)
Comment (ostrokach) [inline examples/pylint/C0321_multiple_statements.py]: Sorry, missed this one!

File: examples/assigning_to_self_example.py (hunk 1 above)
Comment (ostrokach) [inline examples/assigning_to_self_example.py]: Sorry, this was a typo...

File: examples/pylint/W1303_missing_format_argument_key.py (hunk 6 above)
Comment (ostrokach) [inline examples/pylint/W1303_missing_format_argument_key.py]: The arguments were wrapped on a new line indented with 4 spaces to comply with the 80 char limit. This was undone.

File: examples/pylint/R0915_too_many_statements.py (hunk 5 above)
Comment (ostrokach) [inline examples/pylint/R0915_too_many_statements.py]: Removed.

Should I enclose the function name with backticks (`` ` ``) or angle brackets (` < `)?

File: examples/pylint/R0101_too_many_nested_blocks.py [hunk 7]
```diff
@@ -1,6 +1,12 @@
-def cross_join(x_list, y_list, z_list):
-    """Perform an all-by-all join of all elements in the input lists,
-    skipping elements which are None.
+from typing import List, Tuple, TypeVar, Optional
+
+T = TypeVar('T')
+
+def cross_join(x_list: List[Optional[T]], y_list: List[Optional[T]],
```
Comment (david-yz-liu) [inline examples/pylint/R0101_too_many_nested_blocks.py]: Please remove the typevar here (beyond 108/148), and just make this deal with ints.

**Comment by david-yz-liu** [comment]
@ostrokach I merged this in, but there was one change I still want you to make.
